File: springDriven/src/main/java/ru/doronin/fibonacci/springdriven/controllers/FibonacciController.java
package ru.doronin.fibonacci.springdriven.controllers;

import lombok.RequiredArgsConstructor;
import lombok.extern.slf4j.Slf4j;
import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.PathVariable;
import org.springframework.web.bind.annotation.RestController;
import reactor.core.publisher.Mono;
import ru.doronin.fibonacci.springdriven.services.FibonacciCalculationService;

import java.math.BigInteger;

@Slf4j
@RestController
@RequiredArgsConstructor
public class FibonacciController {
    private final FibonacciCalculationService calculationService;

    /**
     * Расчёт элемента ряда Фибоначи
     *
     * @param element - номер элемента
     */
    @GetMapping("/fib/{element}")
    public Mono<BigInteger> calculateElement(@PathVariable("element") long element) {
        return calculationService.calculateElement(element);
    }
}

File: micronaut-driven/src/main/kotlin/ru/doronin/controllers/FibonacciController.kt
package ru.doronin.controllers

import io.micronaut.http.annotation.Controller
import ru.doronin.services.CalculationService
import java.math.BigInteger
import javax.ws.rs.GET
import javax.ws.rs.Path
import javax.ws.rs.PathParam


/* Created by doronantonin on 2021-06-22 */
@Controller
@Path("/fib")
class FibonacciController(private val calculationService: CalculationService) {

    @GET
    @Path("{element}")
    fun calculateElement(@PathParam("element") elementNumber: Long = 1): BigInteger =
            calculationService.calculateElement(elementNumber)
}
